Apply fixes suggested by pre-commit hooks

mdeweerd opened the pull request:
# Apply fixes suggested by pre-commit hooks

Apply some fixes not applied in other PR's

mdeweerd commented:
Applying uncrustify on the sources results in this difference in the test.

It's strange that a file in src would be formatted differently from the other project source files (except thos for tests).

So I am awaiting feed back on this (revert the changes to ChunkStack, or this will be changed so that the test source is copied to the tets directory).

 ```console
diff --git a/home/runner/work/uncrustify/uncrustify/src/ChunkStack.h b/home/runner/work/uncrustify/uncrustify/build/tests/results/src/ChunkStack.h
index b7f4313..6ab1c86 100644
--- a/home/runner/work/uncrustify/uncrustify/src/ChunkStack.h
+++ b/home/runner/work/uncrustify/uncrustify/build/tests/results/src/ChunkStack.h
@@ -16,20 +16,23 @@ class ChunkStack
 public:
    struct Entry
    {
-      Entry() : m_seqnum(0),
-         m_pc(Chunk::NullChunkPtr)
+      Entry()
+         : m_seqnum(0)
+         , m_pc(Chunk::NullChunkPtr)
       {
       }
 
 
-      Entry(const Entry &ref) : m_seqnum(ref.m_seqnum),
-         m_pc(ref.m_pc)
+      Entry(const Entry &ref)
+         : m_seqnum(ref.m_seqnum)
+         , m_pc(ref.m_pc)
       {
       }
 
 
-      Entry(size_t sn, Chunk * pc) : m_seqnum(sn),
-         m_pc(pc)
+      Entry(size_t sn, Chunk *pc)
+         : m_seqnum(sn)
+         , m_pc(pc)
       {
       }
 ```

gmaurel commented:
I will correct this.

mdeweerd commented:
The changes here are mostly whitespace changes, but also uncrustify applied to some of the project's files that are not under the tests directory.

guy-maurel commented:
Yes, it is not the same:
uncrustify -c forUncrustifySources.cfg -f ChunkStack.h -l C
uncrustify -c forUncrustifySources.cfg -f ChunkStack.h -l CPP

You are using the first one, with the language C, which doesn't know what is "class",
The project use the second one.
The file ChunkStack.cpp includes ChunkStack.h, the compiler uses CPP as language.

mdeweerd commented:
> Yes, it is not the same: uncrustify -c forUncrustifySources.cfg -f ChunkStack.h -l C uncrustify -c forUncrustifySources.cfg -f ChunkStack.h -l CPP

I updated the pre-commit configuration (#4494) and adjusted the file ChunkStack.h accordingly.
I added an include for deque as well:
```log
src/ChunkStack.h:45:  Add #include <deque> for deque<>  [build/include_what_you_use] [4]
```

micheleCTDEAdmin commented:
The commit message can be simplified, currently it looks a bit odd.

mdeweerd commented:
I've split the PR in multiple commits with updated descriptions.

micheleCTDEAdmin reviewed (approved):
It looks good
